quality_control_mrp/models/mrp_production.py

```diff
@@ -30,6 +30,8 @@ def action_produce(self, production_id, production_qty, production_mode,
             inspection_model = self.env['qc.inspection']
             production = self.browse(production_id)
             for move in production.move_created_ids2:
+                if move.state != 'done':
```

Comment:
Even more, your option can have performance penalty because of the computation of computed fields on the discarded records.